#+STARTUP: indent
#+SETUPFILE: https://fniessen.github.io/org-html-themes/org/theme-readtheorg.setup
#+TITLE:     RNASeq
#+AUTHOR:    John Brestelli
#+DESCRIPTION: RNASeq Data Type
#+OPTIONS: num:nil
#+HTML_HEAD: <style> #content{max-width:1800px;}</style>
#+HTML_HEAD: <style> p{max-width:800px;}</style>
#+HTML_HEAD: <style> li{max-width:800px;}</style
* RNASeq
** Data Acquisition
*** fastq files (NOT Needed if from EBI)
**** SRA
***** Method 1
1. cd to workSpace directory
2. run "getSampleInfoFromSRA.pl ERPXXXXX" which creates a text file ERPXXXXX.samples.txt
3. modify sample names in ERPXXXXX.samples.txt
4. run "createRNASeqDatasetFromTabFile.pl --sample_file ERPXXXXX.samples.txt  --profileset_name "ADD PROFILE SET NAME HERE --is_strand_specific true/false

***** Method 2
1. testSRAFetch.pl --getFastq  SRX590168
2. all files are named the same so need to rename or you will lose their identity (paired end will produce 2 files - readsFor2.fastq)
   - mv readsFor1.fastq mrna.fastq
3. testSRAFetch.pl does not work for paired-end anymore. A new script is checked into SVN for both single-end and paired-end:
   - SingleSRADump.pl --sampleId SRX2751960 --isPairedEnd
4. The download files are named after sampleId (SRX2751960, SRX2751960.paired)



**** Colorspace SOLiD data
1. use fastq-dump -B to download fastq data with ATCG instead of number in the 2nd lines. e.g. fastq-dump -B -A ERR32754 -B | --dump base Formats sequence using base space (default for other than SOLiD)
2. For colorspace SOLiD data, we can directly use "rnaSeqSample_QuerySRA" dataset class. CBIL/Util/lib/perl/Sra.pm will extract colorspace file to fastq data with ATCG. e.g. [[https://cbilsvn.pmacs.upenn.edu/svn/apidb/ApiCommonDatasets/trunk/Datasets/lib/xml/datasets/FungiDB/anidFGSCA4/Sibthorp_Growth_Conditions.xml][Sibthorp_Growth_Conditions]]

**** BAM files from Provider
1. convert BAM to FASTQ. Probably not the best but it is  doable. e.g. bam2fastq -o s1_48.sorted# s1_48.sorted.bam
   - /eupath/data/EuPathDB/manualDelivery/CryptoDB/cparIowaII/rnaSeq/Lippuner/2012-08-24/workSpace/README
2. You can first sort the bam by read id - if there are read pairs, they will be sorted and paired up. However, this can be a problem if there are pairs where one has no read.
   - samtools sort -n map.Toxo1.sorted.bam > sortedToxo1.bam &
3. An alternative bam to fastq script, a bedtools can be used:
   - bamToFastq -i sortedToxo1.bam -fq sortedToxo1.fastq  -fq2 sortedToxo1Pair.fastq) >& sortedToxo1.err &
** Dataset Configuration 
*** Manual Delivery
**** analysisConfig.txt file
- when there are multiple profile set names for one dataset, each profileSet should have a prefix followed by a space, a dash and a space then something that distinguishes them.
  - Example: yew:/eupath/data/EuPathDB/manualDelivery/PlasmoDB/pberANKA/rnaSeq/Waters_Stunnenberg_five_stages/2014-03-14/final/analysisConfig.xml
- if there are replicates, the replicate sample names should be identical while of course the file names are not
- commas are not tolerated in the sample names and will cause a problem on the website
- If one experiment designed for multiple organisms, in the analysisConfig.xml file, for profileSetName, it can use the same value but separate it with a organism abbreviation.
  - Example(s): yew:/eupath/data/EuPathDB/manualDelivery/AmoebaDB/acasNeff/rnaSeq/Trophozoite_transcriptome/2017-12-29/final/analysisConfig.xml
                yew:/eupath/data/EuPathDB/manualDelivery/AmoebaDB/ehisHM1IMSS/rnaSeq/Trophozoite_transcriptome/2017-11-21/final/analysisConfig.xml
- the names can have "+" or "-" but not slashes ("/")
  - In terms of profileSetName: a note from Kathryn "The problem is that when you have commas in profile set names they work ok on component sites, but it breaks the portal because commas are also used as delimiters there.  We had several problems where datasets were QA'd in the components and signed off, but didn't work in the portal. To prevent this, we made the workflow throw and error when it finds a profileset name with a comma. You can safely change these in the analysisConfig - these are only used internally. A displayName, which can have commas, can be added in the presenter."
- If the sample names are changed in the analysisConfig.xml file after an initial run, in most cases all that is required is that the doTranscriptExpression needs to be undone in the workflow and rerun. However, if the samples have replicates, the step normalizeBedGraph also needs to be undone and rerun. This step will do nothing when undone but as it reruns, it will remove the original analyze_XXXXXXXXX_combined/  directories and then remake them with the correct names. This is true even if the replicate relationships are changed. If normalizeBedGraph is not undone, the old combined dirs will remain and the step copyNormalizedBedGraphFiles will fail because it will not find the dirs it needs.




**** fastq files (optional)
- Place into manual delivery fastq files as described by the rnaSeqExperiment dataset class in [[https://github.com/VEuPathDB/EbrcModelCommon/blob/master/Model/lib/xml/datasetClass/classes.xml][classes.xml]]
- NOTE:  file names with #1 and #2 are not a reliable indication of strandedness when isStrandSpecific=true and hasPairedEnds=true
  - sense = forward should not be the .paired file but antisense = reverse should be .paired
- not needed for RNASeq from EBI
*** Dataset XML
**** NOTES
- adaptor library maintenance
  - There is a data folder under DJob/DistribJobTask/data with the currently available Illumina adaptors for both PE and SE chemistry.
    - if the datasets are from experiments where sequencing chemistry other than TruSeq2 (as used in GAII machines) and TruSeq3 (as used by HiSeq and MiSeq machines) we would need to give a custom fasta file of adaptors or add to ours. ***

** Presenter Configuration
*** Injector 
| Property Name                | Specification                                                                                                                                                                                                                   |
|------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| isEuPathDBSite               | true/false; Set false for sites which are not included in eupathdb.org (like FungiDB)                                                                                                                                           |
| isAlignedToAnnotatedGenome   | true/false; If false, no gene queries will be enabled (only gbrowse tracks)                                                                                                                                                     |
| hasMultipleSamples           | true/false; If true, the standard fold change query will be injected.  Also will inject Fold Change by P Value query if "hasFishersExactTestData" is true.                                                                      |
| hasFishersExactTestData      | true/false; Set this to false to turn off the fold change w/ p value query.  Known BUG:  You must set to false for ALL Strand Specific experiments for the time being.                                                          |
| optionalQuestionDescription  | text; This text will be appended to EVERY injected question.  Most often, you should leave this empty.                                                                                                                          |
| showIntronJunctions          | true/false;  Should we add a gbrowse track for this experiment for intron spanning junctions?  Most often this is true, but check w/ biologist or data provider to ensure this is appropriate                                   |
| graphSampleLables            | semi colon delimited list of overrides for graph x axis labels.  Use this to add short names for graph labels.  Most often leave this blank as it will break if the loaded order changes                                        |
| graphBottomMarginSize        | integer; Set this if the bottom margin is too small and truncates the x axis labels.  Leave empty to start.  Increase as needed.                                                                                                |
| graphForceXLabelsHorizontal  | true/false;  You should start w/ this set to "false" and turn it on if desired.  X axis labels are easier to read when they are horizontal.  The graph default will draw them vertically over some preset number of characters. |
| graphXAxisSamplesDescription | text; This text will appear next to the graph on the gene record page and should describe the x axis labels from the graph.                                                                                                     |
| graphColor                   | Choose a color for the graph.  Color HEX is preferred                                                                                                                                                                           |
| graphPriorityOrderGrouping   | integer;  This controls the ordering of the graph on the gene record page.  it can be used to group similar graphs together.                                                                                                    |
| switchStrandsGBrowse         | boolean; Default is false. Set to true if the profileSet name with ‘firststrand’ is for ‘antisense’, and ‘secondstrand’ is for ‘sense’. This will take care of handling GBrowse big wig files correctly.                        |
| switchStrandsProfiles        | boolean; Default is false. Set to true if the profileSet name with ‘firststrand’ is for ‘antisense’, and ‘secondstrand’ is for ‘sense’. This will take care of handling profile graphs and data tables correctly.               |
| isDESeq                      | boolean; Default is false. Set to true for experiments with 2 or more conditions and more bio replicates each                                                                                                                   |
| isDEGseq                     | boolean; Default is false. leave ‘false’ unless we have a specific request from the user. This is because DEGSeq is not very statistically sound as it works with out replication.                                              |
| includeProfileSimilarity     | is set to "true" if you want to have the profileSimilarity Question with this dataset.                                                                                                                                          |
| profileTimeShift             | param only comes to play IF "includeProfileSimilarity" is set to "true".                                                                                                                                                        |

*** Special Considerations for the Dataset Presenter XML

- Any change made to elements in the presenter xml which are not properties of the "Injector" will require a tuning manager run.
- shortAttribution element of the presenter is required.  This is used in the RNA Seq internal search page.
- datasetShortDisplayName is required.  This is used in the for naming steps in strategies AND is used to name the columns in the results summary page
- hasPairedEnds and isStrandSpecific properties are taken from the Dataset xml (nothing is needed in the presenter xml for these)
- for intron junctions, there is some hard coding based on project in the java class which sets the intron size limit.  HostDB=50000;  All other sites are currently = 5000

** QA
*** Automated QA
*** Manual QA
1. [ ] Begin on the Data Sets record page accessible via the data summary tab.
   - [ ] Check all descriptive text including contact information and publications.
2. [ ] Click through links.
   - [ ] There should be a link to the Internal Search Page for RNA Seq.
   - [ ] There should be a link for a sample Gbrowse region
3. [ ] Click through enabled searches from Internal Search Page.  Ensure that the expected searches are showing up.
4. [ ] Run the searches and check that some genes are returned;
5. [ ] Check the column attribute names in the summary results page;  turn them all on to make sure they are working
6. [ ] Click on at least one gene to ensure the graph is showing up at the desired location in the page;
   - [ ] Check the x axis text; Check the x axis labels for truncated text
7. [ ]  From dataset record, click on "Example Graphs"; ensure the example graph looks ok
8. [ ] For strand specific RNA seq, additional QAs to strandedness on website.
   - [ ] QA Gbrowse track
   - [ ] Compare the glyph color of a gene and RNA seq scale coverage on gbrowse track. If the colors are opposite, change the "switchStrandsGBrowse" to true from false. (Should be able to confirm by looking at several genes.)
   - [ ] QA profile graph/data table
   - [ ] Use the following sql query to check the values in the results table. If the "secondstrand" values are much higher than "firststrand" for all protocol app nodes, change the "switchStrandsProfiles" to true from false. (can also confirm by looking at profiles and data tables on several gene pages.)
     #+begin_src sql
       select s.name, nfe.PROTOCOL_APP_NODE_ID, avg(nfe.value)
       from sres.externaldatabase d, SRES.EXTERNALDATABASERELEASE r, study.study s, study.studylink sl, results.nafeatureexpression nfe
       where d.name = 'XXXXX_rnaSeq_RSRC'
       and d.EXTERNAL_DATABASE_ID = r.external_database_id
       and s.EXTERNAL_DATABASE_RELEASE_ID = r.EXTERNAL_DATABASE_RELEASE_ID
       and s.STUDY_ID = sl.study_id
       and sl.PROTOCOL_APP_NODE_ID = nfe.PROTOCOL_APP_NODE_ID
       group by s.name, nfe.PROTOCOL_APP_NODE_ID
       order by s.name, nfe.PROTOCOL_APP_NODE_ID
     #+end_src
9. [ ] Check the red "new" icon shows up next to this experiment on the RNA Seq Evidence page (modify the file apiCommonModel.xml to change the buildNumber to the current one or update build number in dataset history in the presenter xml)


** Patches
*** Strand Specific Error
For stranded RNA seq, if both strands have equal coverage, need to run the following steps:
1. Delete “mappingStats.txt” files of every sample under master/mainresult dir:
   - For example:  rm /eupath/data/EuPathDB/workflows/FungiDB/29/data/cglaCBS138/gsnap/pHChange_NOStress/analyze_*/master/mainresult/mappingStats.txt
2. Delete “normalized” folders of every sample under master/mainresult dir:
   - For example:  rm -fr /eupath/data/EuPathDB/workflows/FungiDB/29/data/cglaCBS138/gsnap/pHChange_NOStress/analyze_*/master/mainresult/normalized
3. Manually run the “normalizeCoverage.pl ”:
   - For example, stranded, paired-end experiment: nohup normalizeCoverage.pl --inputDir /eupath/data/EuPathDB/workflows/FungiDB/29/data/cglaCBS138/gsnap/pHChange_NOStress --topLevelSeqSizeFile /eupath/data/EuPathDB/workflows/FungiDB/29/data/cglaCBS138/gsnap/topLevelSeqSizes.txt --strandSpecific --isPairedEnd > cglaCBS138_pHChange_NOStress.log 2> cglaCBS138_pHChange_NOStress.log&
4. Run workflow to undo and reload one step:
   - Undo cglaCBS138_gsnap_topLevel.pHChange_NOStress_analyze_rnaSeq_experiment.copyNormalizedBedGraphToWebServiceDir  step in workflow
