* Phase 1 (cleanup)
  + ontology based categorization of attributes and tables
    + Clinepi puts most everything under Data Provenance (should probably do the same for mbio and genomics)
    + [[https://github.com/VEuPathDB/EbrcModelCommon/blob/master/Model/lib/wdk/ontology/commonIndividuals.txt][Individuals File For Datasets]]
    [X] Noted in an early meeting that we are already organizing attributes and tables this way.
  + Provider version table should be removed from Datset Record page
    + set scope=record_internal in individuals.txt to hide
    [X] Fixed wtih EbrcModelCommon [[https://github.com/VEuPathDB/EbrcModelCommon/commit/7b78b72d2e85be8fb7a8d63c6fd61c7cbf7a5cb1#diff-4f17aa4ebc0e1bc949f01be4c328ffce03b1356bcd404b3336ff10907cd5c99d][commit]].
  + hide datasets which have many organisms (example: Pubmed from ncbi;  linkouts)
    + keep the DatasetPresenter xml element for the linkouts because the display_name is used by gene record page
    + could do this by modifying the query "AllDatasets" in [[https://github.com/VEuPathDB/EbrcModelCommon/blob/master/Model/lib/wdk/model/questions/queries/datasetQueries.xml][datasetQueries.xml]] ;  could simply exclude 'Link out' or come up with better way of figuring out which datasets we want to exclude
      #+BEGIN_SRC sql
	select dataset_presenter_id as dataset_id
	from apidbTuning.DatasetPresenter
	where category != 'Link out'
      #+END_SRC
    [X] See commits [[https://github.com/VEuPathDB/EbrcModelCommon/commit/7036f05a65a3aac1c227a316833341010260d4a0#diff-555172edd4681d4e3065a9b67f4de7570ccb80df512ae1f21420dbab00764faf][1]], [[https://github.com/VEuPathDB/EbrcModelCommon/commit/1be55f20506fa5564af4d094d941a120b4c07b80#diff-555172edd4681d4e3065a9b67f4de7570ccb80df512ae1f21420dbab00764faf][2]].
  + pfal3D7_Hagai_ECAssociations_RSRC why is the category "Function Prediction" and not "Link outs"
  + bug on mbio sample record page.  "Description" attribute was showing up at bottom of page.
    [X] See [[https://github.com/VEuPathDB/MicrobiomeModel/commit/698f5d0260034f95ab5fa52a8b8f460b1ecbf97e][commit]]
  + JohnI clean up "HyperLinks" v "HyperLink".
    + rename "HyperLinks" to "GenomicsHyperLink" and provide an additional field to indicate whether the row is internal (based on macro) or external.  
    [X] See [[https://github.com/VEuPathDB/ApiCommonModel/blob/master/Model/lib/xml/tuningManager/apiTuningManager.xml][commit]]
  + Why is the type/category for "%primary_genome%" datasets  "Annotation, curation and identifiers"
    #+BEGIN_SRC sql
     select distinct category
     from apidbtuning.datasetpresenter
     where name like '%primary_genome%';
    #+END_SRC
  
* Phase 2 (organism vs dataset; genomic sites)
   + Existing Organism Record
     + take interesting things from organism page (Gene Types and Sequence Types Tables) and move those to the dataset record page
     [X] Both tables moved.
     + organism results summary page should not link to organism record.  should link to the dataset record page
     []
   + Dataset Record
     + Need a mechanism to show/hide attributes and tables based on the category/type of page
       + We do something like this already on the GeneRecord page.  The gene page only shows attributes and tables which are appropriate for the organism
       + May not need something so fancy.  May be able to modify the Dataset jsx
       + "Example Graphs" table should only show up on RNASeq, Microarray, Phenotype dataset pages
       [X] Pruning tree mechanism implemented. Example Graphs confirmed showing on RNASeq, Microarray, but not genome dataset pages.
     + Genome Datasets
       + Better display name (currenlty they are all called "Genome Sequence and Annotation").  need to include the strain name
       [-] Italics possible for display name.
       + Update primary_genome category?
       + "Source Version" and "Dataset Version" attributes are meaningless for genome datasets.  should not show these attributes for genome datasets
       [X] Source Version and Dataset Version removed in Record.js
       + create attributes for current genome version and current structural annotation version and functional annotation version
       [-] Are we happy with them in the Dataset history table?
       + The current DatasetHistory Table has version fields specific for genomes.  this table should continue to be shown for genome datasets as is
       [X] Made two tables, one for genomes and one for non-genomes. The non-genome one is called DatasetHistory and could possibly simply replace History table.
       + Add "Gene Types" and "Sequence Types" Tables currently shown on Organism Record page
       [X] Added.
       + Add XRefs Table.  We want to show which xref datasets are available for this organism dataset. 
	 + union of xrefs from ebi core database AND xrefs from reflow datasets
     + Hyperlinks
       + Current table called "Relevant pages at PlasmoDB and external resources"
         + Change display name to "External Resources"
	 + For Genomic Sites modify sql to show only hyperlink rows which do not contain Macro
	 + use renamed table for Genomics Sites (should have a field indicating whether the row "is_internal"
         #+BEGIN_SRC sql
	   SELECT dataset_id, text, url, description
	   FROM ApidbTuning.GenomicsDatasetHyperLink
           WHERE is_internal  = 0
         #+END_SRC
       + Current table called "Explore this dataset" currently only shows the wdk Searches for this dataset
	 + Change this to UNION searches and internal hyperlinks
	 + The Datset jsx does something fancy for the searches to get the display name for the search from the wdk.
	 + [[https://github.com/VEuPathDB/EbrcWebsiteCommon/blob/master/Client/src/components/records/DatasetRecordClasses.DatasetRecordClass.jsx][DatasetRecordClasses.DatasetRecordClass.jsx]]
     [X] See EbrcModelCommon [[https://github.com/VEuPathDB/EbrcModelCommon/pull/5][PR]] and EbrcWebsiteCommon [[https://github.com/VEuPathDB/EbrcWebsiteCommon/pull/55][PR]], [[https://github.com/VEuPathDB/EbrcWebsiteCommon/commit/88661d882c3176c713dc0882024282cdcfc4174c][commit]] for all Hyperlinks, Reference table updates.
* Phase 3 (how does a dataset relate to study/investigation?)
   + currenlty in mbio and clinepi we are only showing datasets which are studies (there may be other datasets taxonomy, ontologies, who standards)
   + How to find related studies/datasets;  need some mechanism for grouping datasets
   + Think about how to tell the user when samples are used in multiple assays (could be cross component mbio/clinepi clinepi/plasmodb)
   + How does this relate to the EDA?
     + The Dataset Record will link to the EDA page / study
     + A dataset can be either a study or investigation (multiple related studies)

* PDB Example
  + There are 2 kinds of datasets for PDB Sequences
    + PDBProteinSequences :: sequences downloaded and loaded into tables; our proteins are aligned to these
      + should continue to provide dataset page for pdb sequences
    + _dbxref_pdb_from_annotation_RSRC :: xrefs included in genome annotation
      + hide dbxref / Link out datasets from dataset list.
      + Create new table which lists all xrefs per organism for display on the Genome Dataset Page
  + Each dataset is configured in dataset xml files.  For Genomic sites these files are found in [[https://github.com/VEuPathDB/ApiCommonDatasets/tree/master/Datasets/lib/xml/datasets][ApiCommonDatasts/Datasets/lib/xml/datasets]]
    + PDBProteinSequences dataset is in "global.xml"
    + datasets conform to a dataset Class found in [[https://github.com/VEuPathDB/EbrcModelCommon/blob/master/Model/lib/xml/datasetClass/classes.xml][classes.xml]]
  + Each dataset is required to be handled in Presenter xml Files (either visible datasetPresenter elements or hidden internalDataset elements).  For Genomic Sites the presenter files are found in [[https://github.com/VEuPathDB/ApiCommonPresenters/tree/master/Model/lib/xml/datasetPresenters][ApiCommonPresenters/Model/lib/xml/datasetPresenters]] and [[https://github.com/VEuPathDB/ApiCommonModel/blob/master/Model/lib/xml/datasetPresenters/global.xml][ApiCommonModel/Model/lib/xml/datasetPresenters/global.xml]]
    #+BEGIN_SRC sql
      select *
      from apidbtuning.datasetpresenter
      where name = 'PDBProteinSequences_RSRC'
    #+END_SRC
    
* Queries for genome sequence / annotation
  + Transcript Type Counts
    #+begin_src sql
     select dsp.dataset_presenter_id as dataset_id
		, ta.gene_type
	   , ta.transcript_type
	   , count(*) as transcript_count
      from apidbtuning.organismattributes oa
	 , apidbtuning.transcriptattributes ta
	 , apidbtuning.datasetpresenter dsp
	 , apidbtuning.datasetnametaxon dnt
      where oa.component_taxon_id = ta.taxon_id
      and oa.project_id = ta.project_id
      and oa.component_taxon_id = dnt.taxon_id
      and dnt.dataset_presenter_id = dsp.dataset_presenter_id
      and dsp.type = 'genome'
      and dsp.dataset_presenter_id = 'DS_b08fa4ba88'
      group by oa.project_id, dsp.dataset_presenter_id, ta.gene_type, ta.transcript_type
      order by count(*) desc
    #+end_src

    #+RESULTS: no-hline
      | DS_b08fa4ba88 | protein coding | gene          | mRNA | 13344 |
      | DS_b08fa4ba88 | ncRNA gene     | tRNA          |  305 |       |
      | DS_b08fa4ba88 | ncRNA gene     | pre_miRNA     |   67 |       |
      | DS_b08fa4ba88 | ncRNA gene     | snRNA         |   36 |       |
      | DS_b08fa4ba88 | ncRNA gene     | rRNA          |    9 |       |
      | DS_b08fa4ba88 | ncRNA gene     | SRP_RNA       |    3 |       |
      | DS_b08fa4ba88 | ncRNA gene     | lnc_RNA       |    2 |       |
      | DS_b08fa4ba88 | ncRNA gene     | snoRNA        |    2 |       |
      | DS_b08fa4ba88 | ncRNA gene     | ncRNA         |    2 |       |
      | DS_b08fa4ba88 | ncRNA gene     | RNase_MRP_RNA |    1 |       |
      | DS_b08fa4ba88 | ncRNA gene     | RNase_P_RNA   |    1 |       |

  + Sequence Type Counts
    #+begin_src sql
      select dsp.dataset_presenter_id, sa.sequence_type, fl.feature_type, count(*) as feature_count
	   from apidbtuning.organismattributes oa
	 , ApidbTuning.GenomicSeqAttributes sa
	 , apidb.featurelocation fl
	 , apidbtuning.datasetpresenter dsp
	 , apidbtuning.datasetnametaxon dnt
      where oa.component_taxon_id = sa.taxon_id
      and oa.project_id = sa.project_id
      and sa.na_sequence_id = fl.na_sequence_id
      and oa.component_taxon_id = dnt.taxon_id
      and dnt.dataset_presenter_id = dsp.dataset_presenter_id
      and dsp.type = 'genome'
      and dsp.dataset_presenter_id = 'DS_b08fa4ba88'
      group by dsp.dataset_presenter_id, sa.sequence_type, fl.feature_type
      order by count(*) desc
    #+end_src
    
    #+RESULTS: no-hline
     | DS_b08fa4ba88 | supercontig | LowComplexityNAFeature | 439121 |
     | DS_b08fa4ba88 | supercontig | Repeats                | 155973 |
     | DS_b08fa4ba88 | supercontig | TandemRepeatFeature    | 102066 |
     | DS_b08fa4ba88 | supercontig | TransposableElement    |  61046 |
     | DS_b08fa4ba88 | supercontig | ExonFeature            |  55917 |
     | DS_b08fa4ba88 | supercontig | CDS                    |  54114 |
     | DS_b08fa4ba88 | supercontig | Intron                 |  43480 |
     | DS_b08fa4ba88 | supercontig | UTR                    |  18081 |
     | DS_b08fa4ba88 | supercontig | Transcript             |  13772 |
     | DS_b08fa4ba88 | supercontig | GeneFeature            |  13578 |
     | DS_b08fa4ba88 | supercontig | ScaffoldGapFeature     |  11084 |
     | DS_b08fa4ba88 | supercontig | RNAFeature             |    353 |

  + Other Data Associated with this Genome
    #+begin_src sql

	select dsp.dataset_presenter_id, o_dsp.category, count(*) as dataset_count
	  from apidbtuning.organismattributes oa
	     , apidbtuning.datasetpresenter o_dsp
	     , apidbtuning.datasetnametaxon o_dnt
	     , apidbtuning.datasetnametaxon dnt
	     , apidbtuning.datasetpresenter dsp
	  where oa.component_taxon_id = o_dnt.taxon_id
	  and o_dnt.dataset_presenter_id = o_dsp.dataset_presenter_id
	  and oa.component_taxon_id = dnt.taxon_id
	  and dnt.dataset_presenter_id = dsp.dataset_presenter_id
	  and dsp.type = 'genome'
    and dsp.dataset_presenter_id = 'DS_b08fa4ba88'
	  group by dsp.dataset_presenter_id, o_dsp.category
    order by count(*) desc
    #+end_src

    #+RESULTS: no-hline
       | DS_b08fa4ba88 | RNASeq                               | 5 |
       | DS_b08fa4ba88 | Link outs                            | 2 |
       | DS_b08fa4ba88 | Genetic variation                    | 2 |
       | DS_b08fa4ba88 | Immunology                           | 1 |
       | DS_b08fa4ba88 | Transcriptomics                      | 1 |
       | DS_b08fa4ba88 | Annotation, curation and identifiers | 1 |

  + External Databases/Resources

    #+begin_src sql
      select dsp.dataset_presenter_id
	   , ext.dataset as name
      from apidbtuning.datasetnametaxon dnt, apidbtuning.datasetpresenter dsp,

      (select distinct * from (
      SELECT 
	edd.dataset_presenter_display_name AS dataset
	,ga.taxon_id
      FROM
	 sres.dbref db
      , DOTS.dbrefnafeature dbna
      , apidbtuning.ExternalDbDatasetPresenter edd
      , sres.externaldatabaserelease edr
      , ApidbTuning.geneAttributes ga
      WHERE
	db.external_database_release_id = edd.external_database_release_id
	AND edr.external_database_release_id = edd.external_database_release_id
	AND dbna.db_ref_id = db.db_ref_id
	AND ga.na_feature_id = dbna.na_feature_id
	UNION
      SELECT 
	edd.dataset_presenter_display_name AS dataset
	, ta.taxon_id
      FROM
	 sres.dbref db
      , DOTS.dbrefnafeature dbna
      , apidbtuning.ExternalDbDatasetPresenter edd
      , sres.externaldatabaserelease edr
      , ApidbTuning.transcriptAttributes ta
      WHERE
	db.external_database_release_id = edd.external_database_release_id
	AND edr.external_database_release_id = edd.external_database_release_id
	AND dbna.db_ref_id = db.db_ref_id
	AND ta.na_feature_id = dbna.na_feature_id
	UNION
      SELECT d.name as dataset
      , ga.taxon_id
      FROM
	 sres.dbref dbr
      , DOTS.dbrefnafeature dbrf
      , sres.externaldatabaserelease r
      , sres.externaldatabase d
      , ApidbTuning.geneAttributes ga
      , APIDB.EXTERNALRESOURCEURL eru
      WHERE dbr.external_database_release_id = r.external_database_release_id
      and r.external_database_id = d.external_database_id
      and dbr.db_ref_id = dbrf.db_ref_id
      and dbrf.na_feature_id = ga.na_feature_id
      and upper(d.name) = eru.database_name
      UNION
      SELECT d.name as dataset
      , ta.taxon_id
      FROM
	 sres.dbref dbr
      , DOTS.dbrefaafeature dbrf
      , Dots.aafeature aaf
      , sres.externaldatabaserelease r
      , sres.externaldatabase d
      , ApidbTuning.transcriptAttributes ta
      , APIDB.EXTERNALRESOURCEURL eru
      WHERE dbr.external_database_release_id = r.external_database_release_id
      and r.external_database_id = d.external_database_id
      and dbr.db_ref_id = dbrf.db_ref_id
      and dbrf.aa_feature_id = aaf.aa_feature_id
      and aaf.aa_sequence_id = ta.aa_sequence_id
      and upper(d.name) = eru.database_name
      )
      ) ext
      where ext.taxon_id = dnt.taxon_id
      and dnt.dataset_presenter_id = dsp.dataset_presenter_id
      and dsp.type = 'genome'
      and dsp.dataset_presenter_id = 'DS_b08fa4ba88'
      order by ext.dataset
    #+end_src

    #+RESULTS: no-hline
     | DS_b08fa4ba88 | EMBL              |
     | DS_b08fa4ba88 | HAMAP             |
     | DS_b08fa4ba88 | KEGG_Enzyme       |
     | DS_b08fa4ba88 | PFAM              |
     | DS_b08fa4ba88 | PIRSF             |
     | DS_b08fa4ba88 | PRINTS            |
     | DS_b08fa4ba88 | RFAM              |
     | DS_b08fa4ba88 | SMART             |
     | DS_b08fa4ba88 | SUPERFAMILY       |
     | DS_b08fa4ba88 | TIGRFAM           |
     | DS_b08fa4ba88 | UniParc           |
     | DS_b08fa4ba88 | Uniprot/SPTREMBL  |
     | DS_b08fa4ba88 | Uniprot/SWISSPROT |
     | DS_b08fa4ba88 | protein_id        |


  + EBI Dataset Mappings (some dataset names were updated because EBI data load.  those should still be searchable)

    #+begin_src sql
      select dataset_presenter_id, name
           , 'DS_' || lower(substr(standard_hash(replace(name, '_ebi_', '_'), 'SHA1'), 0, 10))  as previous_dataset_id
           , replace(name, '_ebi_', '_') as previous_name
      from apidbtuning.datasetpresenter where name like '%_ebi_%'
    #+end_src

    #+RESULTS: no-hline
     | DS_aae60e6bf7 | pfal3D7_Tonkin_Hill_Malaria_ebi_rnaSeq_RSRC             | DS_601b15d9f7 | pfal3D7_Tonkin_Hill_Malaria_rnaSeq_RSRC             |
     | DS_9c91e450ac | pfal3D7_Josling_Schizont_Transcriptomes_ebi_rnaSeq_RSRC | DS_044c11419a | pfal3D7_Josling_Schizont_Transcriptomes_rnaSeq_RSRC |
     | DS_b9eaf20251 | pyoeyoelii17X_Kappe_ebi_rnaSeq_RSRC                     | DS_998f7032ed | pyoeyoelii17X_Kappe_rnaSeq_RSRC                     |
     | DS_a239464cb4 | pfal3D7_Stunnenberg_ebi_rnaSeq_RSRC                     | DS_d57671ced8 | pfal3D7_Stunnenberg_rnaSeq_RSRC                     |
