#+STARTUP: indent
#+SETUPFILE: https://fniessen.github.io/org-html-themes/org/theme-readtheorg.setup
#+TITLE:     DNASeq
#+AUTHOR:    John Brestelli
#+DESCRIPTION: DNASeq
#+OPTIONS:   H:5 num:nil toc:2 p:t tags:not-in-toc
* B53 and B54
** [[readtheorgs:/project_planning/unidb.org][UniDB]]
* IN_PROGRESS DNASeq Nextflow workflow
** Input Data
- [X] Can be fastq files or BAM files
- [X] All data from SRA will be aligned by EBI and given as bam
- [X] The nextflow workflow will operate on an experiment. not a single sample
** SNP Calling
- [X] move over varscan steps into nextflow workflow
- [X] Do not use Strain name in VCF file. how do we uniquely identify a strain  
** Down Sampling
- [X] filter reads if too many;  add configuration option for setting max
- [X] recalcuate mapping stats after down sampling
** CNV 
 - [X] move over DJob/DistribJobTasks/bin/runCNVTasks.pl
 - [X] LOH data for plot
** Indels
- [X] Include Indels in VCF File
- [ ] Publish indel vcf file
** Consensus Sequence
- [X] use additional characters in consensus seq for ambiguous bases

* Test Oracle Database
- [X] load subset of genome
- [X] rm43647

* TODO DNASeq Reflow Workflow
No longer have GFF files per sample;  We have VCF file per Experiment;  Need to merge experiment vcf into a Per Organism VCF

** Indels
- [ ] Store indels in database table
- [ ] New Feature table for Indel
- [ ] These are stored in Ref Seq coords

** Process Sequence Variations Step
- [ ] read indels so we can calculate products better
- [ ] load indels so we can show consensus sequence better
- [ ] use IUPAC (R=A or G) characteres in consensus sequence...  but BioPerl does not handle these correctly when translating codon to aa/product

** Additional Tables and upgrade to ApiDB.SNP
- [ ] ApiDB.SNPAlleleSummary
- [ ] ApiDB.SNPProductSummary
- [ ] new fields to ApiDB.SNP
  - [ ] REF_CODON
  - [ ] POSITION_IN_CODON
  - [ ] ALLELE_TOTAL
  - [ ] HAS_STOP_CODON
  - [ ] for minor allele should use the second most abundant allele only
  - [ ] incorporate variations somehow (JSON Array? or VCF clob?)

** VCFs
- [ ] Merge a VCF file per organism
- [ ] Run SNPEff (or VEP)
- [ ] copy VCF file to apiSiteFiles

** HSSS WS Files
- [ ] Regenerate these from VCF
  
** Sample Details
- [ ] loader for eda schema (entity graph)
- [ ] will we use the eda or filter params?

* Model
** Improvements to Variation Page
- [ ] Incorporate new tables
- [ ] Better show sample details (not just "Country" table... should be able to show all sample details
- [ ] EDA?
** Searches
- [ ] new dynamic columns for High Speed SNP Searches?
** Results
- [ ] Send results to JBrowse as Track;  Requires filtering the VCF;  Does this impact Loading???
** JBrowse
- [ ] VCF track
- [ ] SNPEff Track

* Consensus Sequence
- [ ] Alignment tool on frontend needs to incorporate indels;
  - [ ] need a count of all insertions and all deletions
  - [ ] Indels are stored in "Reference Space"

