<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2021-09-07 Tue 12:57 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>&lrm;</title>
<meta name="generator" content="Org mode" />
<meta name="author" content="John Brestelli" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<link rel="stylesheet" type="text/css" href="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/htmlize.css"/>
<link rel="stylesheet" type="text/css" href="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/readtheorg.css"/>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script type="text/javascript" src="https://fniessen.github.io/org-html-themes/src/lib/js/jquery.stickytableheaders.min.js"></script>
<script type="text/javascript" src="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/js/readtheorg.js"></script>
<script type="text/javascript">
// @license magnet:?xt=urn:btih:e95b018ef3580986a04669f1b5879592219e2a7a&dn=public-domain.txt Public Domain
<!--/*--><![CDATA[/*><!--*/
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
    /*]]>*///-->
// @license-end
</script>
</head>
<body>
<div id="content">
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#org0950f32">Redmine PROJECT</a>
<ul>
<li><a href="#org7a8fd03">UniDB</a>
<ul>
<li><a href="#org8f0107a">Description</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#org9aa31a0">B53</a>
<ul>
<li><a href="#org11a134e"><span class="done DONE">DONE</span> Site Configuration</a></li>
<li><a href="#org39e961e"><span class="done DONE">DONE</span> Workflow Changes</a></li>
<li><a href="#org7e397b3"><span class="done DONE">DONE</span> Schema Changes</a></li>
<li><a href="#org1aad71d"><span class="done DONE">DONE</span> ApiCommonDatasts</a></li>
<li><a href="#org81f7e85"><span class="done DONE">DONE</span> Plugin Changes for Sequence Variations</a></li>
<li><a href="#org21fbba0"><span class="done DONE">DONE</span> Tuning Manager</a></li>
<li><a href="#org22e8c97"><span class="done DONE">DONE</span> Model</a></li>
<li><a href="#orgd53d3a1"><span class="done DONE">DONE</span> FeatureLocation Patch</a></li>
<li><a href="#org7535bb9"><span class="done DONE">DONE</span> MassSpecFeature Patch</a></li>
<li><a href="#orgb1696dd">Punt</a></li>
</ul>
</li>
<li><a href="#orge11c98e">B54</a>
<ul>
<li><a href="#orgdcb91c0"><span class="done DONE">DONE</span> Genomics Workflow Changes</a></li>
<li><a href="#orgde70ad6"><span class="todo TODO">TODO</span> Schema Changes</a></li>
<li><a href="#org396bdfb"><span class="done DONE">DONE</span> Bulk Loader for SNP / Seqvars</a></li>
<li><a href="#orgc6df0f6"><span class="done DONE">DONE</span> Add Mass Spec profile (Llinas) redmine:45318</a></li>
<li><a href="#orgba0eb29"><span class="todo TODO">TODO</span> Update SRes.Taxon and SRes.TaxonName with most recent  version of NCBI Taxonomy</a></li>
<li><a href="#orgb8d375d"><span class="done DONE">DONE</span> Undo unidbcomponents</a></li>
</ul>
</li>
<li><a href="#orge42e46f">Future Releases</a>
<ul>
<li><a href="#orgf20f2d7">Plugin Changes for DatabaseMapping fields</a></li>
<li><a href="#org154444d">Tuning Manager</a>
<ul>
<li><a href="#orgdc66d38">Why so many dependencies on DatasetPresenter?</a></li>
<li><a href="#org81fdb9b">Intron Junctions redmine:45717</a></li>
<li><a href="#orgcd6842f">Move Organism Specific to workflow</a></li>
<li><a href="#org6c9851b">Comparative Genomics ???</a></li>
</ul>
</li>
<li><a href="#org2d3d6d2">Model (JBrowse, WDK, WS Plugins, &#x2026;)</a></li>
<li><a href="#org158d34f">wgcna (host /pathogen)</a></li>
</ul>
</li>
<li><a href="#org5869c9b">Beta Release</a>
<ul>
<li><a href="#org02a5c3f">Automated QA Process</a></li>
</ul>
</li>
<li><a href="#orgeb41f45">SOP for schema changes</a></li>
<li><a href="#org61bf1f4">Use Cases For Unified Database</a></li>
<li><a href="#orgede820d">Basics</a></li>
<li><a href="#org9bcd924">Table Reader(s)</a></li>
<li><a href="#org5dc4fdf">ebi2GUS</a></li>
</ul>
</div>
</div>
<div id="outline-container-org0950f32" class="outline-2">
<h2 id="org0950f32">Redmine PROJECT</h2>
<div class="outline-text-2" id="text-org0950f32">
</div>
<div id="outline-container-org7a8fd03" class="outline-3">
<h3 id="org7a8fd03">UniDB</h3>
<div class="outline-text-3" id="text-org7a8fd03">
<p>
<a href="https://redmine.apidb.org/projects/jbdqh">Project</a>
<a href="https://redmine.apidb.org/projects/jbdqh/issues/new">New Issue</a>
</p>
</div>
<div id="outline-container-org8f0107a" class="outline-4">
<h4 id="org8f0107a">Description</h4>
<div class="outline-text-4" id="text-org8f0107a">
<div class="DESCRIPTION" id="org7ca0dcd">
<p>

</p>

</div>
</div>
</div>
</div>
</div>


<div id="outline-container-org9aa31a0" class="outline-2">
<h2 id="org9aa31a0">B53</h2>
<div class="outline-text-2" id="text-org9aa31a0">
</div>
<div id="outline-container-org11a134e" class="outline-3">
<h3 id="org11a134e"><span class="done DONE">DONE</span> Site Configuration</h3>
<div class="outline-text-3" id="text-org11a134e">
<ul class="org-ul">
<li class="on"><code>[X]</code> (Cristina) confiugre beta and qa sites to use "uni-build-53" and uni053n</li>
</ul>
</div>
</div>
<div id="outline-container-org39e961e" class="outline-3">
<h3 id="org39e961e"><span class="done DONE">DONE</span> Workflow Changes</h3>
<div class="outline-text-3" id="text-org39e961e">
<ul class="org-ul">
<li class="on"><code>[X]</code> Remove step copy to webservices and download files</li>
</ul>
</div>
</div>
<div id="outline-container-org7e397b3" class="outline-3">
<h3 id="org7e397b3"><span class="done DONE">DONE</span> Schema Changes</h3>
<div class="outline-text-3" id="text-org7e397b3">
<ul class="org-ul">
<li class="on"><code>[X]</code> study.study study<sub>id</sub> and investigation<sub>id</sub> need to be number(8)</li>
<li class="on"><code>[X]</code> Add "database<sub>orig</sub>" and "primary<sub>key</sub><sub>orig</sub>" fields to every table in "Core.TableInfo"</li>
<li class="on"><code>[X]</code> Add Index on each table for 3 fields (pk, pk<sub>orig</sub>, db<sub>orig</sub>)</li>
<li class="on"><code>[X]</code> <p>
new script to add fields after apidbschema has been made.  last script when creating apidb schema
</p>
<div class="org-src-container">
<pre class="src src-sql">BEGIN
FOR item in (select di.name || '.' || di.name as table_name
		  , ti.primary_key_column
		  , di.name || '_' || di.name || '_map' as index_name
		  from (-- everything but version and userdataset schemas
		  select  t.*
		  from core.tableinfo t, core.databaseinfo d
		  where lower(t.table_type) != 'version'
		  and t.DATABASE_ID = d.DATABASE_ID
		  and d.name not in ('UserDatasets', 'ApidbUserDatasets', 'chEBI', 'hmdb')
		  and t.name not in ('AlgorithmParam','GlobalNaturalKey','DatabaseTableMapping','SnpLinkage', 'CompoundPeaksChebi')
		  minus
		  -- minus Views on tables
		  select * from core.tableinfo where view_on_table_id is not null
		  ) ti, core.databaseinfo di
		  where ti.database_id = di.database_id   )
LOOP
 ALTER table item.table_name add database_orig varchar2(30), primary_key_orig number(20);
 CREATE INDEX index_name ON item.table_name (database_orig, primary_key_orig, item.primary_key_column) tablespace indx; 
END LOOP;
END;
</pre>
</div></li>
</ul>
</div>
</div>
<div id="outline-container-org1aad71d" class="outline-3">
<h3 id="org1aad71d"><span class="done DONE">DONE</span> ApiCommonDatasts</h3>
<div class="outline-text-3" id="text-org1aad71d">
<ul class="org-ul">
<li class="on"><code>[X]</code> configure only dbs which have been exported for b53</li>
<li class="on"><code>[X]</code> bld ApiCommonDatasts and generateFromDatasets</li>
</ul>
</div>
</div>
<div id="outline-container-org81f7e85" class="outline-3">
<h3 id="org81f7e85"><span class="done DONE">DONE</span> Plugin Changes for Sequence Variations</h3>
<div class="outline-text-3" id="text-org81f7e85">
<ul class="org-ul">
<li class="on"><code>[X]</code> Skip ApiDB.SequenceVariation
<ul class="org-ul">
<li><p>
~ 58 hours to load 633 mil rows from Plasmo (fungi has &gt; 1.3 billion rows)
</p>
<pre class="example" id="org0e63c50">
Sun Apr  4 18:22:17 2021	Begin ID Lookup for ApiDB::SequenceVariation from database plas051n
Sun Apr  4 18:24:35 2021	Finished ID Lookup for ApiDB::SequenceVariation from database plas051n
Sun Apr  4 18:29:19 2021	Processed 100000 from ApiDB::SequenceVariation
Sun Apr  4 18:29:55 2021	Processed 200000 from ApiDB::SequenceVariation
....
Wed Apr  7 05:30:14 2021	Processed 633000000 from ApiDB::SequenceVariation
Wed Apr  7 05:30:44 2021	Processed 633100000 from ApiDB::SequenceVariation
Wed Apr  7 05:30:48 2021	Finished Reading data from ApiDB::SequenceVariation
Wed Apr  7 05:30:48 2021	Finished Loading 633110728 Rows into table ApiDB::SequenceVariation from database plas051n
</pre></li>
</ul></li>
</ul>
</div>
</div>
<div id="outline-container-org21fbba0" class="outline-3">
<h3 id="org21fbba0"><span class="done DONE">DONE</span> Tuning Manager</h3>
<div class="outline-text-3" id="text-org21fbba0">
<ul class="org-ul">
<li class="on"><code>[X]</code> SNPAttributes (ok for b53 but will be missing some fields)</li>
<li class="on"><code>[X]</code> <p>
SNPStrains (update sql so it doesn't use SequenceVariation)
</p>
<div class="org-src-container">
<pre class="src src-sql">select * from study.protocolappnode where name like '% (Sequence Variation)'
</pre>
</div></li>
</ul>
</div>
</div>
<div id="outline-container-org22e8c97" class="outline-3">
<h3 id="org22e8c97"><span class="done DONE">DONE</span> Model</h3>
<div class="outline-text-3" id="text-org22e8c97">
<ul class="org-ul">
<li class="on"><code>[X]</code> Using include/exclude projects make the snpattributes which are missing internal for b53</li>
</ul>
</div>
</div>
<div id="outline-container-orgd53d3a1" class="outline-3">
<h3 id="orgd53d3a1"><span class="done DONE">DONE</span> FeatureLocation Patch</h3>
<div class="outline-text-3" id="text-orgd53d3a1">
<ul class="org-ul">
<li class="on"><code>[X]</code> <p>
Truncate apidb.featurelocation
</p>
<div class="org-src-container">
<pre class="src src-sql">truncate table apidb.featurelocation;
</pre>
</div></li>
<li class="on"><code>[X]</code> remove orf code from GUSTableLoader</li>
<li class="on"><code>[X]</code> <p>
make new Temporary apidb.databasetablemapping (decode inc-&gt;053n) and make indexe and delete rows from apidb.databasetablemapping for featurelocation
</p>
<div class="org-src-container">
<pre class="src src-sql">create table apidb.databasetablemappingPatch as 
select database_table_mapping_id
     , decode(database_orig, 'cryp-inc', 'cryp053n',
			     'giar-inc', 'giar053n',
			     'ameb-inc', 'ameb053n',
			     'piro-inc', 'piro053n',
			     'toxo-inc', 'toxo053n',
			     'vect-inc', 'vect053n',
			     'host-inc', 'host053n',
			     'tryp-inc', 'tryp053n',
			     'tvag-inc', 'tvag053n',
			     'fung-inc', 'fung053n',
			     'plas-inc', 'plas053n',
			     'micr-inc', 'micr053n', 'ERROR') as database_orig
     , table_name
     , primary_key_orig
     , primary_key
     , modification_date
     , user_read
     , user_write
     , group_read
     , group_write
     , other_read
     , other_write
     , row_user_id
     , row_group_id
     , row_project_id
     , row_alg_invocation_id
from apidb.databasetablemapping
where table_name !=  'ApiDB::FeatureLocation';
</pre>
</div></li>
<li class="on"><code>[X]</code> <p>
rename table (log in as apidb user)
</p>
<div class="org-src-container">
<pre class="src src-sql">rename databasetablemapping to databasetablemapping_old
rename databasetablemappingpatch to databasetablemapping
</pre>
</div></li>
<li class="on"><code>[X]</code> <p>
add indexes
</p>
<div class="org-src-container">
<pre class="src src-sql">GRANT INSERT, SELECT, UPDATE, DELETE ON ApiDB.DatabaseTableMapping TO gus_w;
GRANT SELECT ON ApiDB.DatabaseTableMapping TO gus_r;
ALTER TABLE ApiDB.DatabaseTableMapping ADD CONSTRAINT database_map_tab2_pk PRIMARY KEY (database_table_mapping_id);
CREATE INDEX apidb.db_tbl_map_tab2_idx ON ApiDB.DatabaseTableMapping (database_orig, table_name, primary_key_orig, primary_key) tablespace indx;
</pre>
</div></li>
<li class="on"><code>[X]</code> <p>
Run plugin manually for FeatureLocation (can we use direct path mode?)
</p>
<div class="org-src-container">
<pre class="src src-bash">ga ApiCommonData::Load::Plugin::InsertUniDB --database piro053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/piro --mode load --commit ;
ga ApiCommonData::Load::Plugin::InsertUniDB --database ameb053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/ameb --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database cryp053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/cryp --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database giar053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/giar --mode load --commit ;
ga ApiCommonData::Load::Plugin::InsertUniDB --database host053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/host --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database micr053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/micr --mode load --commit ;
ga ApiCommonData::Load::Plugin::InsertUniDB --database plas053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/plas --mode load --commit ;
ga ApiCommonData::Load::Plugin::InsertUniDB --database toxo053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/toxo --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database tvag053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/tvag --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database tryp053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/tryp --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database fung053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/fung --mode load --commit;
ga ApiCommonData::Load::Plugin::InsertUniDB --database vect053n --table_reader 'ApiCommonData::Load::GUSTableReader' --logDir /home/jbrestel/tmp/featureLocationPatch/vect --mode load --commit;
</pre>
</div></li>
<li class="on"><code>[X]</code> decode 053n -&gt; inc</li>
</ul>
</div>
</div>
<div id="outline-container-org7535bb9" class="outline-3">
<h3 id="org7535bb9"><span class="done DONE">DONE</span> MassSpecFeature Patch</h3>
<div class="outline-text-3" id="text-org7535bb9">
<ul class="org-ul">
<li class="on"><code>[X]</code> <p>
update string6 to source<sub>id</sub> to retain orig source<sub>id</sub> 
</p>
<div class="org-src-container">
<pre class="src src-sql">update dots.aafeatureimp 
set string6 = source_id
where subclass_view = 'MassSpecFeature';;
</pre>
</div></li>
<li class="on"><code>[X]</code> use mapping table to update source<sub>id</sub>:  yew:~/patches/patchMassSpecFeature.pl</li>
<li class="on"><code>[X]</code> Patch uni-inc</li>
</ul>
</div>
</div>
<div id="outline-container-orgb1696dd" class="outline-3">
<h3 id="orgb1696dd">Punt</h3>
<div class="outline-text-3" id="text-orgb1696dd">
<ol class="org-ol">
<li>SequenceVariations were not loaded;
<ul class="org-ul">
<li>tables on snp record page</li>
<li>one or two snp attributes</li>
</ul></li>
<li>Intron Junctions
<ul class="org-ul">
<li>jbrowse track (turn off)</li>
</ul></li>
<li>User Datasets (installed into one app db only)</li>
<li>Llinas Mass Spec Profiles</li>
</ol>
</div>
</div>
</div>
<div id="outline-container-orge11c98e" class="outline-2">
<h2 id="orge11c98e">B54</h2>
<div class="outline-text-2" id="text-orge11c98e">
</div>
<div id="outline-container-orgdcb91c0" class="outline-3">
<h3 id="orgdcb91c0"><span class="done DONE">DONE</span> Genomics Workflow Changes</h3>
<div class="outline-text-3" id="text-orgdcb91c0">
<ul class="org-ul">
<li class="on"><code>[X]</code> Add other 2 SNP Attributes which are being added by tuning manager</li>
<li class="on"><code>[X]</code> Required rebuild for all snp datasets</li>
</ul>
</div>
</div>
<div id="outline-container-orgde70ad6" class="outline-3">
<h3 id="orgde70ad6"><span class="todo TODO">TODO</span> Schema Changes</h3>
<div class="outline-text-3" id="text-orgde70ad6">
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> use perl script to automatically add "database<sub>orig</sub>" and "primary<sub>key</sub><sub>orig</sub>" in installer</li>
<li class="on"><code>[X]</code> add 3 new fields to SRes.OntologySynonym</li>
</ul>
</div>
</div>
<div id="outline-container-org396bdfb" class="outline-3">
<h3 id="org396bdfb"><span class="done DONE">DONE</span> Bulk Loader for SNP / Seqvars</h3>
</div>
<div id="outline-container-orgc6df0f6" class="outline-3">
<h3 id="orgc6df0f6"><span class="done DONE">DONE</span> Add Mass Spec profile (Llinas) <a href="https://redmine.apidb.org/issues/45318">redmine:45318</a></h3>
</div>
<div id="outline-container-orgba0eb29" class="outline-3">
<h3 id="orgba0eb29"><span class="todo TODO">TODO</span> Update SRes.Taxon and SRes.TaxonName with most recent  version of NCBI Taxonomy</h3>
</div>
<div id="outline-container-orgb8d375d" class="outline-3">
<h3 id="orgb8d375d"><span class="done DONE">DONE</span> Undo unidbcomponents</h3>
</div>
</div>
<div id="outline-container-orge42e46f" class="outline-2">
<h2 id="orge42e46f">Future Releases</h2>
<div class="outline-text-2" id="text-orge42e46f">
</div>
<div id="outline-container-orgf20f2d7" class="outline-3">
<h3 id="orgf20f2d7">Plugin Changes for DatabaseMapping fields</h3>
<div class="outline-text-3" id="text-orgf20f2d7">
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> Database Mapping Table
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> Global Rows still need to populate existing Database Mapping Table</li>
<li class="off"><code>[&#xa0;]</code> Component Specific Rows should populate the 2 new fields in the primary table</li>
<li class="off"><code>[&#xa0;]</code> Undo needs to work BOTH ways because of legacy data in genomics sites</li>
<li class="off"><code>[&#xa0;]</code> Existing queries using databasemapping table need to be union of both ways because of legacy data in genomics sites</li>
</ul></li>
<li class="off"><code>[&#xa0;]</code> Address cleanup steps for failures
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> Database Specific / Non Global we get for free as there is no clean up needed after above change</li>
<li class="off"><code>[&#xa0;]</code> <p>
For Global Rows, we must delete the apidb.databasetablemapping rows for this table + alg<sub>invocation</sub>
</p>
<div class="org-src-container">
<pre class="src src-sql">delete apidb.databasetablemapping where database_orig = ? and table_name = ? and row_alg_invocation_id = ?
</pre>
</div></li>
<li class="off"><code>[&#xa0;]</code> <p>
update how we find unidb aware tables
</p>
<div class="org-src-container">
<pre class="src src-sql">select ti.name as table_name
     , di.name as database_name
     , ti.primary_key_column
from core.tableinfo ti, core.databaseinfo di,
     (select owner, table_name
     from all_tab_columns 
     where column_name in ('DATABASE_ORIG', 'PRIMARY_KEY_ORIG')
     and owner != 'EDA' -- TEMPORARY
     group by owner, table_name
     having count(*) = 2) wit
where ti.database_id = di.database_id
and upper(wit.owner) = upper(di.name)
and upper(wit.table_name) = upper(ti.name)
and ti.VIEW_ON_TABLE_ID is null
</pre>
</div></li>
</ul></li>
</ul>
</div>
</div>
<div id="outline-container-org154444d" class="outline-3">
<h3 id="org154444d">Tuning Manager</h3>
<div class="outline-text-3" id="text-org154444d">
</div>
<div id="outline-container-orgdc66d38" class="outline-4">
<h4 id="orgdc66d38">Why so many dependencies on DatasetPresenter?</h4>
</div>
<div id="outline-container-org81fdb9b" class="outline-4">
<h4 id="org81fdb9b">Intron Junctions <a href="https://redmine.apidb.org/issues/45717">redmine:45717</a></h4>
</div>
<div id="outline-container-orgcd6842f" class="outline-4">
<h4 id="orgcd6842f">Move Organism Specific to workflow</h4>
</div>
<div id="outline-container-org6c9851b" class="outline-4">
<h4 id="org6c9851b">Comparative Genomics ???</h4>
</div>
</div>
<div id="outline-container-org2d3d6d2" class="outline-3">
<h3 id="org2d3d6d2">Model (JBrowse, WDK, WS Plugins, &#x2026;)</h3>
<div class="outline-text-3" id="text-org2d3d6d2">
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> Profile Similarity (Inject this instead of hard coding)</li>
<li class="off"><code>[&#xa0;]</code> Phenotype (Injection)</li>
<li class="off"><code>[&#xa0;]</code> Cellular Location (TriTyp, Giardia?) (Injection)</li>
<li class="off"><code>[&#xa0;]</code> MassSpec
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> Change tuning manager and model to use mass<sub>spec</sub><sub>summary</sub><sub>id</sub> in place of source<sub>id</sub> (grep dots.massspecfeature)</li>
<li class="off"><code>[&#xa0;]</code> Confirm with data loaders (update ApiCommonData)</li>
</ul></li>
<li class="off"><code>[&#xa0;]</code> remove project<sub>id</sub> from webservice queries (once we remove federated portal)</li>
<li class="off"><code>[&#xa0;]</code> Text Searches (will site search fix these?)</li>
</ul>
</div>
</div>
<div id="outline-container-org158d34f" class="outline-3">
<h3 id="org158d34f">wgcna (host /pathogen)</h3>
</div>
</div>

<div id="outline-container-org5869c9b" class="outline-2">
<h2 id="org5869c9b">Beta Release</h2>
<div class="outline-text-2" id="text-org5869c9b">
</div>
<div id="outline-container-org02a5c3f" class="outline-3">
<h3 id="org02a5c3f">Automated QA Process</h3>
<div class="outline-text-3" id="text-org02a5c3f">
<ol class="org-ol">
<li class="on"><code>[X]</code> compare list of wdk Searches, Attributes, and Tables in UniDB to union of all component projects
<ul class="org-ul">
<li class="on"><code>[X]</code> Record expected and Actual Results</li>
</ul></li>
<li class="on"><code>[X]</code> Selenium Tests for Search Pages (Bindu)</li>
<li class="on"><code>[X]</code> use the webserive queries for Performance Tests
<ul class="org-ul">
<li class="on"><code>[X]</code> compare plasmodb.org (b52) to feature.veupathdb.org</li>
<li class="on"><code>[X]</code> address faiures</li>
<li class="on"><code>[X]</code> Record expected and Actual Results</li>
</ul></li>
<li class="on"><code>[X]</code> attribute tables
<ul class="org-ul">
<li class="on"><code>[X]</code> using sql minus operation, compare 052 attribute tables in eupa052n to uni-inc</li>
<li class="on"><code>[X]</code> address faiures (JohnI)</li>
<li class="on"><code>[X]</code> Record expected and Actual Results</li>
</ul></li>
<li class="on"><code>[X]</code> Record page tables;  can test against individual components
<ul class="org-ul">
<li class="on"><code>[X]</code> foreach record table, run the sql and compare counts (union for all components)</li>
<li class="on"><code>[X]</code> address faiures</li>
<li class="on"><code>[X]</code> Record expected and Actual Results</li>
</ul></li>
<li class="on"><code>[X]</code> Run all Param queries on uni052n vs union of components
<ul class="org-ul">
<li class="on"><code>[X]</code> Deal with enum params</li>
<li class="on"><code>[X]</code> is there a webservice way to get the values for all params</li>
</ul></li>
</ol>
</div>
</div>
</div>
<div id="outline-container-orgeb41f45" class="outline-2">
<h2 id="orgeb41f45">SOP for schema changes</h2>
<div class="outline-text-2" id="text-orgeb41f45">
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> How to handle schema changes?
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> add columns</li>
<li class="off"><code>[&#xa0;]</code> add tables</li>
<li class="off"><code>[&#xa0;]</code> alter size of columns</li>
</ul></li>
</ul>
</div>
</div>
<div id="outline-container-org61bf1f4" class="outline-2">
<h2 id="org61bf1f4">Use Cases For Unified Database</h2>
<div class="outline-text-2" id="text-org61bf1f4">
<ul class="org-ul">
<li>fully functional portal (veupathdb.org)</li>
<li>host / pathogen / vector</li>
<li>user defined organism preferences</li>
<li>Fewer prod db instances</li>
<li>ebi2gus</li>
</ul>
</div>
</div>
<div id="outline-container-orgede820d" class="outline-2">
<h2 id="orgede820d">Basics</h2>
<div class="outline-text-2" id="text-orgede820d">
<ul class="org-ul">
<li><a href="https://docs.google.com/document/d/1K3ckE6hwN9r-Dp1Av_zDH5Jcr7ApbBjQ-7yJM1zk0bQ/edit">UniDB Design Document December 2019</a></li>
<li><a href="https://wiki.apidb.org/index.php/UniDB%20Workflow">Wiki / Technical info about running workflow</a></li>
</ul>
</div>
</div>
<div id="outline-container-org9bcd924" class="outline-2">
<h2 id="org9bcd924">Table Reader(s)</h2>
<div class="outline-text-2" id="text-org9bcd924">
<ul class="org-ul">
<li><a href="file:///home/jbrestel/project_home/ApiCommonData/Load/lib/perl/UniDBTableReader.pm">UniDB Table Reader</a>
<ul class="org-ul">
<li>initial thinking was that sql queries would map input-&gt;output</li>
</ul></li>
<li><a href="file:///home/jbrestel/project_home/ApiCommonData/Load/lib/perl/GUSTableReader.pm">GUSTableReader</a>
<ul class="org-ul">
<li>inpput is a GUS oracle instance</li>
<li>queries here are like "select * from $table"</li>
</ul></li>
<li><a href="file:///home/jbrestel/project_home/ApiCommonData/Load/lib/perl/EBITableReader.pm">EBI Table Reader</a>
<ul class="org-ul">
<li>input is flat files</li>
<li>no queries.  the translation from chado/mysql-&gt;GUS is done via <a href="https://github.com/VEuPathDB/ebi2gus/">ebi2gus</a> and uses the E! perl API
<ul class="org-ul">
<li><a href="file:///home/jbrestel/project_home/ebi2gus/Dockerfile">file:///home/jbrestel/project_home/ebi2gus/Dockerfile</a></li>
<li><a href="file:///home/jbrestel/project_home/ebi2gus/lib/perl/EBIParser.pm">file:///home/jbrestel/project_home/ebi2gus/lib/perl/EBIParser.pm</a></li>
<li><a href="file:///home/jbrestel/project_home/ebi2gus/lib/perl/GUS/DoTS/GeneFeature.pm">file:///home/jbrestel/project_home/ebi2gus/lib/perl/GUS/DoTS/GeneFeature.pm</a></li>
</ul></li>
</ul></li>
</ul>
</div>
</div>

<div id="outline-container-org5dc4fdf" class="outline-2">
<h2 id="org5dc4fdf">ebi2GUS</h2>
<div class="outline-text-2" id="text-org5dc4fdf">
<ul class="org-ul">
<li>given mysql dump (init.sql), the container will fire up mariadb server, run the init.sql, will run dumpGUS.pl which creates a set of files (one file per GUS Table)</li>
<li>new branch needed
<ul class="org-ul">
<li>change to GUS schema ( ~ 40 tables for "core genome")</li>
<li>changes to input schema</li>
<li>changes to the perl api</li>
</ul></li>
</ul>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="author">Author: John Brestelli</p>
<p class="date">Created: 2021-09-07 Tue 12:57</p>
<p class="validation"><a href="https://validator.w3.org/check?uri=referer">Validate</a></p>
</div>
</body>
</html>
